ui/org.eclipse.pde.ui.tests/src/org/eclipse/pde/ui/tests/target/IUBundleContainerTests.java

```diff
@@ -327,9 +377,58 @@ public void testExternalModelManagerPreferences() throws Exception {
 	}
 
 	/**
-	 * Creates an IU bundle container with the specified IUs from the test repository.
+	 * Creates an in-memory artifact repository over {@code test/sites/site.a.b}
 	 *
-	 * @param unitIds identifiers of IU's to add to the container
+	 * @return in-memory artifact repository
+	 */
+	protected VirtualArtifactRepository createVirtualArtifactRepository() throws Exception {
+		File repoFolder = new File(getURI("/tests/sites/site.a.b"));
+		// DirectoryBundleContainer expects features to be unpacked
+		File features = new File(repoFolder, "features");
+		for (File jarFile : features.listFiles()) {
+			if (jarFile.isDirectory()) {
+				continue;
+			}
+			String jarFileName = jarFile.getName();
+			String jarFileBaseName = jarFileName.substring(0, jarFileName.length() - ".jar".length());
+
+			File outDir = new File(features, jarFileBaseName);
+			outDir.mkdir();
+			outDir.deleteOnExit();
+
+			try (JarFile jar = new JarFile(jarFile)) {
```

Comment:
This entire block just extract the jar into a temporary directory, doesn't it?
I'm relatively sure, that there are (probably multiple) implementations of this functionality in some File-Utility classes in PDE (core) that can be reused here.